tests/unit/Storage/Hydration/EventHydratorTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Krixon\DomainEvent\TestUnit\Storage\Hydration;
+
+use Krixon\DateTime\DateTime;
+use Krixon\DomainEvent\Storage\Hydration\EventHydrator;
+use Krixon\DomainEvent\Storage\Hydration\Exception\HydrationFailed;
+use Krixon\DomainEvent\Storage\Hydration\Resolver\EventClassResolver;
+use Krixon\DomainEvent\TestResource\Event\ItemRenamedEvent;
+use PHPUnit\Framework\TestCase;
+
+class EventHydratorTest extends TestCase
+{
+    public function testCanHydrateEvent() : void
+    {
+        $classResolver = $this->createMock(EventClassResolver::class);
+
+        $classResolver
+            ->method('resolve')
+            ->with('item.renamed')
+            ->willReturn('Krixon\DomainEvent\TestResource\Event\ItemRenamedEvent');
+
+        $hydrator = new EventHydrator($classResolver);
+
+        $occurredOn = DateTime::now();
+
+        $data = [
+            'eventType'    => 'item.renamed',
+            'eventVersion' => 5,
+            'occurredOn'   => $occurredOn,
```

Comment:
@krixon we could handle the `occuredOn` property explicitly by converting it to / from a `string` format within the event hydrator. Alternatively we could go a step further by introducing the concept of type converters that can be injected into the hydrator itself which would be more flexible, supporting any value object. Thoughts?

Reply:
I think that's more the responsibility of whatever is working with this normalised array. For example, if this is serialised to JSON the date needs to be in a certain format which might not be right for other cases.